File: Blind 75 LeetCode/Dhruva Bhat/Two Sum/Two Sum.java
class Solution {
    public int[] twoSum(int[] A, int k) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int i = 0, a;; i++){
            if(map.containsKey(A[i]))
                return new int[]{i, map.get(A[i])};
            map.put(k - A[i], i);
        }
    }
}

File: Blind 75 LeetCode/Sriragavi P/Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> res;
        unordered_map<int, int> m;
        
        for(int i = 0; i < nums.size(); i++) {
            int toFind = target - nums[i];
            if(m.count(toFind) && m[toFind] != i) {
                res.push_back(i);
                res.push_back(m[toFind]);
                return res;
            }
            m[nums[i]] = i;
        }
        
        return res;
            
    }
};